src/test/java/com/casper/sdk/helper/CasperTransferHelperTests.java

```diff
@@ -78,6 +81,70 @@ void testTransferOnTestnet() throws IOException, NoSuchTypeException, GeneralSec
     }
 
+    @Test
+    void testTransferOnNctl() throws IOException, NoSuchTypeException, GeneralSecurityException, URISyntaxException, ValueSerializationException {
+        Ed25519PrivateKey user1 = new Ed25519PrivateKey();
+        Ed25519PrivateKey user2 = new Ed25519PrivateKey();
+        user1.readPrivateKey(getResourcesKeyPath("deploy-accounts/nctl/users/user-1/secret_key.pem"));
+        user2.readPrivateKey(getResourcesKeyPath("deploy-accounts/nctl/users/user-2/secret_key.pem"));
+
+        long id = Math.abs(new Random().nextInt());
+        Ttl ttl = Ttl
+                .builder()
+                .ttl("30m")
+                .build();
+        Random rnd = new Random();
+        boolean coin = rnd.nextBoolean();
+        Ed25519PrivateKey from;
+        PublicKey to;
+        if (coin) {
+            from = user1;
+            to = PublicKey.fromAbstractPublicKey(user2.derivePublicKey());
+        } else {
+            from = user2;
+            to = PublicKey.fromAbstractPublicKey(user1.derivePublicKey());
+        }
+
+        Deploy deploy = CasperTransferHelper.buildTransferDeploy(from, to,
+                BigInteger.valueOf(2500000000L), "casper-net-1",
+                id, BigInteger.valueOf(100000000L), 1L, ttl, new Date(),
+                new ArrayList<>());
+        DeployResult deployResult = casperServiceNctl.putDeploy(deploy);
+        assertEquals(deployResult.getDeployHash(), Hex.toHexString(deploy.getHash().getDigest()));
+    }
+
+    @Test
+    void testSpeculativeTransferOnNctl() throws IOException, NoSuchTypeException, GeneralSecurityException, URISyntaxException, ValueSerializationException {
+        Ed25519PrivateKey user1 = new Ed25519PrivateKey();
+        Ed25519PrivateKey user2 = new Ed25519PrivateKey();
+        user1.readPrivateKey(getResourcesKeyPath("deploy-accounts/nctl/users/user-1/secret_key.pem"));
+        user2.readPrivateKey(getResourcesKeyPath("deploy-accounts/nctl/users/user-2/secret_key.pem"));
+
+        long id = Math.abs(new Random().nextInt());
+        Ttl ttl = Ttl
+                .builder()
+                .ttl("30m")
+                .build();
+        Random rnd = new Random();
+        boolean coin = rnd.nextBoolean();
+        Ed25519PrivateKey from;
+        PublicKey to;
+        if (coin) {
+            from = user1;
+            to = PublicKey.fromAbstractPublicKey(user2.derivePublicKey());
+        } else {
+            from = user2;
+            to = PublicKey.fromAbstractPublicKey(user1.derivePublicKey());
+        }
+
+        Deploy deploy = CasperTransferHelper.buildTransferDeploy(from, to,
+                BigInteger.valueOf(2500000000L), "casper-net-1",
+                id, BigInteger.valueOf(100000000L), 1L, ttl, new Date(),
+                new ArrayList<>());
+        SpeculativeDeployData deployData = speculativeCasperServiceNctl.speculativeExec(deploy);
+        assertNotNull(deployData);
```

Comment:
I'd like to see the SpeculativeDeployData executionResult etc is populated with valid results in this test too. We'll need a working node to test it though.